### PR TITLE
nshlib: fix top command sorting result error

### PR DESCRIPTION
**Pull request by Gary-Hobson**
## Summary
nsh: Add top command and watch command

## Testing
sim

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



**No, this PR does not meet the NuttX requirements.**

**Missing Information:**

* **Summary:**
    *  Lacks a description of why the change is necessary. Is it a new feature request? Does it fix a bug? 
    *  Needs more detail on how the "top" and "watch" commands work. What information do they display? How are they used?
* **Impact:** 
    *  All impact sections are empty.  You must address each point (user impact, build impact, hardware impact, etc.) with either "NO" or a "YES" followed by a description.
* **Testing:**
    *  "sim" is not enough information. Specify the  host OS, CPU, compiler, target architecture, board, and configuration.
    *  Provide actual testing logs from before and after the change.

**To improve this PR:**

1. **Expand the Summary:** Clearly articulate the purpose and functionality of the new commands.
2. **Complete the Impact Section:** Assess and describe the impact of your changes on all aspects mentioned.
3. **Provide Detailed Testing Information:** List your testing environment specifics and include relevant logs demonstrating the changes' effects.

File: system/telnetd/Kconfig [hunk 1]
```diff
@@ -40,6 +40,6 @@ config SYSTEM_TELNETD_SESSION_PRIORITY
 
 config SYSTEM_TELNETD_SESSION_STACKSIZE
 	int "Telnetd session task stack size"
-	default 3072
+	default SYSTEM_NSH_STACKSIZE
```
Comment (xiaoxiang781216) [inline system/telnetd/Kconfig:43]: drop the patch which has been upstreamed

Comment (Gary-Hobson) [reply system/telnetd/Kconfig]: Done

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
@Gary-Hobson please fix ci error:
```
====================================================================================
Configuration/Tool: same70-xplained/mcuboot-loader,CONFIG_ARM_TOOLCHAIN_GNU_EABI
2024-10-14 16:17:34
------------------------------------------------------------------------------------
  Cleaning...
  Configuring...
  Disabling CONFIG_ARM_TOOLCHAIN_GNU_EABI
  Enabling CONFIG_ARM_TOOLCHAIN_GNU_EABI
  Building NuttX...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

  2 1669k    2 36869    0     0  66926      0  0:00:25 --:--:--  0:00:25 66926
100 1669k  100 1669k    0     0  2001k      0 --:--:-- --:--:-- --:--:-- 5772k
arm-none-eabi-ld: /github/workspace/sources/nuttx/nuttx section `.ramfunc' will not fit in region `flash'
arm-none-eabi-ld: region `flash' overflowed by 52 bytes
arm-none-eabi-ld: warning: /github/workspace/sources/nuttx/nuttx has a LOAD segment with RWX permissions
make[1]: *** [Makefile:212: nuttx] Error 1
make: *** [tools/Unix.mk:551: nuttx] Error 2
make: Target 'all' not remade because of errors.
/github/workspace/sources/nuttx/tools/testbuild.sh: line 385: /github/workspace/sources/nuttx/../nuttx/nuttx.manifest: No such file or directory
  Normalize same70-xplained/mcuboot-loader
On branch master
Your branch is up to date with 'origin/master'.

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	boards/arm/samv7/same70-xplained/scripts/flash.ld.tmp

nothing added to commit but untracked files present (use "git add" to track)
```

**Comment by Gary-Hobson** [comment]
> @Gary-Hobson please fix ci error:
> 
Enable LTO in the following PR to avoid insufficient flash errors during compilation
https://github.com/apache/nuttx/pull/14287

**Comment by xiaoxiang781216** [comment]
@Gary-Hobson please fix the conflict.

**Pull request merged**